### DISCUSSION TITLE
Higher fractional second precision

### DISCUSSION THREAD

**Discussion by moorereason** [pelletier/go-toml#707] [Questions] [answered]
### The Situation

While doing differential fuzzing against [`toml.dart`](https://github.com/just95/toml.dart), I had a failed case where `go-toml` rejected the input with an error while `toml.dart` accepted the input as-is:

```
--- FAIL: FuzzUnmarshal (3173.97s)
    --- FAIL: FuzzUnmarshal (0.01s)
        fuzz_test.go:174: output errors differ:
            input:
                "0=00:00:00.0000000000"
            /x/bin/gotoml-test-decoder (outErr=true):
            Error decoding TOML: toml: maximum precision for date time is nanosecond

            Exit 1

            /x/src/github.com/just95/toml.dart/build/bin/decoder (outErr=false):
            {"0":{"type":"time-local","value":"00:00:00.000000000000"}}
```

### The TOML spec

> Millisecond precision is required. Further precision of fractional seconds is implementation-specific. If the value contains greater precision than the implementation can support, the additional precision must be truncated, not rounded.

### The Question

Precision beyond milliseconds is implementation specific.  Should `go-toml` return an error here or truncate the fractional second to the maximum supported precision (nanoseconds) and move on?

### My Two Cents

This input is valid TOML.  It seems like `go-toml` should parse the document as best it can, but we should make clear in the README and/or godocs that we're limited to nanosecond precision on the fractional second component and will truncate as necessary.

**Comment by pelletier** [comment, marked as answer]
Agreed. I think we should truncate to nanoseconds and note that behavior in godoc. I didn’t realize the spec was prescriptive on what to do for greater than supported precision. I think it’s fair to use nanoseconds as the supported precision because that’s what the standard behavior does. If somebody needs greater precision They can use their own type with custom un/marshaling methods.

**Reply by pelletier** [reply]
https://github.com/pelletier/go-toml/pull/710
